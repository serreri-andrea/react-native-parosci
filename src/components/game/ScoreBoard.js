/**
 * Created by Andrea on 18/03/2018.
 */

import React, { Component }     from 'react';
import {
    StyleSheet,
    Text,
    View
} from 'react-native';
import Rules                    from "../../lib/Rules";
import Localization             from "../../localization/LocalizedStrings";
import Colors                   from "../../theme/colors";

export default class ScoreBoard extends Component{

    constructor(props) {
        super(props);
    }

    getWinner(){
        let winner = Rules.getVictory(this.props.mode, this.props.playerOneCard,this.props.playerTwoCard );
        if (winner === 1){
            return (Localization.getStringOfKey("playerA"))
        }else if (winner === 2){
            return (Localization.getStringOfKey("playerB"))
        }else{
            return(Localization.getStringOfKey("draw"))
        }
    }


    renderMatchInfo(){
        if (this.props.playerOneCard && this.props.playerTwoCard) {
            return (
                <View style={{alignItems:'center'}}>
                    <Text style={styles.title}>{Localization.getStringOfKey("playerA")} {Localization.getStringOfKey("played")}:
                        <Text style={styles.res}> {Localization.getStringOfKey(this.props.playerOneCard).toUpperCase() || ""}</Text>
                    </Text>
                    <Text style={styles.title}>{Localization.getStringOfKey("playerB")} {Localization.getStringOfKey("played")}:
                        <Text style={styles.res}> {Localization.getStringOfKey(this.props.playerTwoCard).toUpperCase() || ""}</Text>
                    </Text>
                    <Text style={styles.title}>{Localization.getStringOfKey("winner")}:
                        <Text style={styles.res}> {this.getWinner()}</Text>
                    </Text>
                </View>
            )
        }else{
            return(<View/>)
        }
    }


    render(){
        let p1Score = this.props.score && this.props.score.p1Score ? this.props.score.p1Score : 0;
        let p2Score = this.props.score && this.props.score.p2Score ? this.props.score.p2Score : 0;
        return(
            <View style={styles.container}>
                <Text style={styles.title}>{Localization.getStringOfKey("score")}:
                    <Text style={styles.res}> {p1Score} - {p2Score}</Text>
                </Text>
                {this.renderMatchInfo()}
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container:{
        alignItems:'center',
        marginTop:10,
        marginBottom:10,
        flex:1,
    },
    title:{
        fontSize:18,
        color: "white",
    },
    res:{
        fontSize:18,
        color:Colors.secondary
    }
});
