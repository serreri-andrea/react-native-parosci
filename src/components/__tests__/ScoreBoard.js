import 'react-native';
import React            from 'react';
import ScoreBoard       from "../game/ScoreBoard";

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly: without props', () => {
    const tree = renderer.create(
        <ScoreBoard/>
    );
    expect(tree).toMatchSnapshot();
});

it('renders correctly: with props', () => {
    const tree = renderer.create(
        <ScoreBoard playerOneCard={"fgyhjuk"} playerTwoCard={"fghjklm"} mode={"fghjklm"} score={"ghjkl"} />
    );
    expect(tree).toMatchSnapshot();
});

it('renders correctly: with props number', () => {
    const tree = renderer.create(
        <ScoreBoard playerOneCard={"paper"} playerTwoCard={"rock"} mode={"classic"} score={{p1Score:0, p2Score:1}} />
    );
    expect(tree).toMatchSnapshot();
});
