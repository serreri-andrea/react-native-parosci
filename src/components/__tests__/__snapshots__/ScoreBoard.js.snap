// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly: with props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     Score is : 
    0
     - 
    0
  </Text>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       You played: 
      fgyhjuk
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       IA plays: 
      fghjklm
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       Winner: 
      Draw
    </Text>
  </View>
</View>
`;

exports[`renders correctly: with props number 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     Score is : 
    0
     - 
    1
  </Text>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       You played: 
      paper
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       IA plays: 
      rock
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
       Winner: 
      You
    </Text>
  </View>
</View>
`;

exports[`renders correctly: without props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "marginBottom": 10,
      "marginTop": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     Score is : 
    0
     - 
    0
  </Text>
  <View />
</View>
`;
